packages/frontend/src/hooks/useSignupWithWallet.ts

```diff
@@ -1,56 +1,35 @@
-import { useCallback, useContext } from "react";
-import { ethers } from 'ethers';
-
 declare global {
     interface Window {
         ethereum: any
     }
 }
 
 const useSignupWithWallet = (
-    hashUserId: string | null, 
-    userContext: any, 
-    setIsLoading: any
+    navigate: (path: string) => void,
+    setIsSignupLoading: (loading: boolean) => void,
+    handleWalletSignMessage: () => Promise<void>,
+    load: () => Promise<void>,
+    signup: () => Promise<void>,
 ) => {
-    const signUpWithWallet = useCallback(async () => {
-
+    const signUpWithWallet = async () => {
         try {
-            setIsLoading(true);
-            if (!hashUserId) {
-                throw new Error('Invalid user')
-            };
+            setIsSignupLoading(true)
             if (!window.ethereum) {
                 throw new Error('請安裝MetaMask錢包')
-            };
-            console.log('waiting sign up...')
-            const accounts = await window.ethereum.request({ method: 'eth_requestAccounts' });
-            const account = accounts[0];
-
-            const signature = await window.ethereum.request({
-                method: 'personal_sign',
-                params: [
-                    ethers.utils.hexlify(
-                        ethers.utils.toUtf8Bytes(hashUserId)
-                    ),
-                    account,
-                ],
-            });
-
-            localStorage.setItem('signature', signature);
-
-            await userContext.load();
-
-            await userContext.signup();
-
+            }
+            navigate('/')
+            handleWalletSignMessage()
+            await load()
+            await signup()
+            console.log('has signed up')
```

Comment:
no need to run `load()` here, and should run `navigate('/')` after signup successfully

Reply:
If didn't load first, userState will not be initialize. But maybe we can modify the flow to make it more logical.
Because they said they want users can read post while signup is progressing, I move it to top.

Reply:
Oh! It's my mistake. I ran load in useUserInit. You are right. Thank you!